add option to explicitly use provided offset, even when 0.

Indra-db opened the pull request:
This allows to register fields in arbitrary order for reflection. This is useful for example where in Rust fields are not guaranteed to be in the same order as represented.

copygirl commented on include/flecs/addons/meta.h:
```diff
@@ -202,6 +202,7 @@ typedef struct EcsMember {
     int32_t count;                                 /**< Number of elements (for inline arrays). */
     ecs_entity_t unit;                             /**< Member unit. */
     int32_t offset;                                /**< Member offset. */
+    bool explicit_offset;                          /**< if Offset should be explicitly used. */
```
Should probably also get the capitalization fix.

Indra-db replied:
yup, noticed when the CI failed haha :D, thanks !

Indra-db commented:
> Looks good! Can you:
> 
> * Add a test that shows the new feature working (out of order member declaration with 0 offset not being the first member)
> * Rename the member to `use_offset`

added test, renamed and fixed `.member` setting the field correctly with overloading.

copygirl commented on include/flecs/addons/meta.h:
```diff
@@ -202,6 +202,7 @@ typedef struct EcsMember {
     int32_t count;                                 /**< Number of elements (for inline arrays). */
     ecs_entity_t unit;                             /**< Member unit. */
     int32_t offset;                                /**< Member offset. */
+    bool use_offset;                          /**< If offset should be explicitly used. */
```
Another nitpick: Alignment is off.

SanderMertens commented:
Thanks for the PR!
